code/modules/synthesized_instruments/song_editor.dm

```diff
@@ -17,15 +17,14 @@
 
 
 /datum/nano_module/song_editor/proc/current_page()
-	return src.song.current_line > 0 ? Ceiling(src.song.current_line / GLOB.musical_config.song_editor_lines_per_page) : src.page
+	return src.song.current_line > 0 ? Ceiling(src.song.current_line / GLOB.musical_config.song_editor_lines_per_page) : min(src.page, pages())
```

Comment:
Seems superficial: if `pages` is decreased, `page` should be modified accordingly instead.